lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -268,7 +268,7 @@ def setup_address_hash(options)
         }
 
         submitted_address = options[:billing_address] || options[:address] || default_address
-        options[:billing_address] = default_address.merge(submitted_address) { |_k, default, submitted| submitted.blank? ? default : submitted }
+        options[:billing_address] = default_address.merge(submitted_address.symbolize_keys) { |_k, default, submitted| submitted.blank? ? default : submitted }
```

Comment:
An alternative would be to create default_address with string keys from the get go.
We could also use `merge!` with the bang, since we instantiate `default_address` on the fly.

Reply:
> An alternative would be to create default_address with string keys from the get go.

That wouldn't work since the symbol accessor is used here https://github.com/activemerchant/active_merchant/blob/e049610610d3e20aa0d76188db200976aea97b08/lib/active_merchant/billing/gateways/cyber_source.rb#L526

> We could also use merge! with the bang, since we instantiate default_address on the fly.

Didn't you told that `merge` was faster?

Reply:
I thought merge! was faster, quickly googled [here](https://www.paperlesspost.com/blog/teams/different-methods-for-merging-ruby-hashes/)
> Using merge! is almost 2 times as fast

Reply:
Is `symbolize_keys` OK to use here? I believe that is a Rails function and not a Ruby function, and this repo does not have Rails as a dependency.

https://apidock.com/rails/Hash/symbolize_keys

Reply:
Ahh thanks - the tests all passed for me but I didn't see how it was being included since ActiveSupport isn't in the Gemfile :+1: